R/iterate.R

```diff
@@ -0,0 +1,118 @@
+#' Perform requests iteratively, generating new requests from previous responses
+#'
+#' @description
+#' `r lifecycle::badge("experimental")`
+#'
+#' `req_perform_iteratively()` iteratively generates and performs requests,
+#' using a callback function, `next_req`, to define the next request based on
+#' the current request and response. You will probably want to it pair with an
+#' [iteration helper][iterate_with_offset] and use a
+#' [multi-response handler][resps_combine] to process the result.
+#'
+#' @inheritParams req_perform
+#' @param next_req A function that takes the previous response (`resp`) and
+#'   request (`req`) and returns a [request] for the next page or `NULL` if
+#'   the iteration should terminate.
```

Comment:
This should mention how to signal the number of pages and we might want to provide a helper for that.

Reply:
Also we might provide an example (including the technique with `known_total`) or point the user to where to look this up.

Reply:
Tracking in #358 — I think I'm likely to move the "do this once" check to `req_perform_iteratively()`.